pkg/resource/replication_group/delta_util.go

```diff
@@ -62,3 +79,74 @@ func engineVersionsMatch(
 
 	return false
 }
+
+// compares desired.Spec to last applied slow log config in latest.Status to determine
+// whether to invoke an update or not. This comes down to only trying to apply a particular configuration
+// once.
+func slowLogRequiresUpdate(desired *resource, latest *resource) bool {
+	var desiredConfig *svcapitypes.LogDeliveryConfigurationRequest
+	var lastRequestedConfig *svcapitypes.LogDeliveryConfiguration
+
+	// retrieve desiredConfig from desired Spec
+	if desired.ko.Spec.LogDeliveryConfigurations != nil {
+		for _, config := range desired.ko.Spec.LogDeliveryConfigurations {
+			if config.LogType != nil && *config.LogType == "slow-log" {
+				desiredConfig = config.DeepCopy()
+			}
+		}
+	}
+
+	// retrieve lastRequestedConfig from latest Status
+	if latest.ko.Status.LastRequestedLogDeliveryConfigurations != nil {
+		for _, config := range latest.ko.Status.LastRequestedLogDeliveryConfigurations {
+			if config.LogType != nil && *config.LogType == "slow-log" {
+				lastRequestedConfig = config.DeepCopy()
+			}
+		}
+	}
+
+	// if both nil, no update to be done.
+	// if desired nil and latest non-nil, then the intention could be to disable; update required.
+	// similarly, if desired non-nil and latest nil, then log delivery must be enabled; update required.
+	// if both non-nil, compare values to determine whether an update is required
+	if desiredConfig == nil && lastRequestedConfig == nil {
+		return false
+	} else if desiredConfig == nil || lastRequestedConfig == nil {
+		return true
+	} else {
+		if !reflect.DeepEqual(desiredConfig.DestinationDetails, lastRequestedConfig.DestinationDetails) {
+			return true
+		}
+
+		if ackcompare.HasNilDifference(desiredConfig.DestinationType, lastRequestedConfig.DestinationType) {
+			return true
+		} else if desiredConfig.DestinationType != nil && lastRequestedConfig.DestinationType != nil {
+			if *desiredConfig.DestinationType != *lastRequestedConfig.DestinationType {
+				return true
+			}
+		}
+
+		// types are different here but we can still compare nil-ness
+		if ackcompare.HasNilDifference(desiredConfig.Enabled, lastRequestedConfig.Status) {
```

Comment:
I think you're missing a case here: If `desiredConfig.Enabled == nil` && `latestRequestConfig == &"false"`.
`ackcompare.HasNilDifference` will return true, but logically speaking they are equal and the function should return `false` instead

Reply:
Noted. I think the new approach with storing the last requested configuration in `Annotations` should solve this.

The current version is a bit of a hack since I am forcing the `LogDeliveryConfigurationsRequest` type into the `LogDeliveryConfigurations` type (since I could not introduce a `LogDeliveryConfigurationsRequest` type into `Status`, as it does not appear in an API response).